lang/kotlin/kotlin.py

```diff
@@ -0,0 +1,182 @@
+
+    def code_operator_subscript():
+        actions.insert("[]")
+        actions.key("left")
+
+    def code_operator_assignment():
+        actions.auto_insert(" = ")
+
+    def code_operator_subtraction():
+        actions.auto_insert(" - ")
+
+    def code_operator_subtraction_assignment():
+        actions.auto_insert(" -= ")
+
+    def code_operator_addition():
+        actions.auto_insert(" + ")
+
+    def code_operator_addition_assignment():
+        actions.auto_insert(" += ")
+
+    def code_operator_multiplication():
+        actions.auto_insert(" * ")
+
+    def code_operator_multiplication_assignment():
+        actions.auto_insert(" *= ")
+
+    def code_operator_exponent():
+        actions.auto_insert(" ^ ")
+
+    def code_operator_division():
+        actions.auto_insert(" / ")
+
+    def code_operator_division_assignment():
+        actions.auto_insert(" /= ")
+
+    def code_operator_modulo():
+        actions.auto_insert(" % ")
+
+    def code_operator_modulo_assignment():
+        actions.auto_insert(" %= ")
+
+    def code_operator_equal():
+        actions.auto_insert(" == ")
+
+    def code_operator_not_equal():
+        actions.auto_insert(" != ")
+
+    def code_operator_greater_than():
+        actions.auto_insert(" > ")
+
+    def code_operator_greater_than_or_equal_to():
+        actions.auto_insert(" >= ")
+
+    def code_operator_less_than():
+        actions.auto_insert(" < ")
+
+    def code_operator_less_than_or_equal_to():
+        actions.auto_insert(" <= ")
+
+    def code_operator_and():
+        actions.auto_insert(" && ")
+
+    def code_operator_or():
+        actions.auto_insert(" || ")
+
+    def code_operator_bitwise_and():
+        actions.auto_insert(" & ")
+
+    def code_operator_bitwise_or():
+        actions.auto_insert(" | ")
+
+    def code_operator_bitwise_exclusive_or():
+        actions.auto_insert(" ^ ")
+
+    def code_operator_bitwise_left_shift():
+        actions.auto_insert(" << ")
+
+    def code_operator_bitwise_left_shift_assignment():
+        actions.auto_insert(" <<= ")
+
+    def code_operator_bitwise_right_shift():
+        actions.auto_insert(" >> ")
+
+    def code_operator_bitwise_right_shift_assignment():
+        actions.auto_insert(" >>= ")
+
+    def code_self():
+        actions.auto_insert("this")
+
+    def code_insert_null():
+        actions.insert("null")
+
+    def code_insert_is_null():
+        actions.insert(" == null")
+
+    def code_insert_is_not_null():
+        actions.insert(" != null")
+
+    def code_state_if():
+        actions.insert("if () ")
+        actions.key("left")
+        actions.key("left")
+
+    def code_state_else_if():
+        actions.insert("else if () ")
+        actions.key("left")
+        actions.key("left")
+
+    def code_state_else():
+        actions.insert("else ")
+        actions.key("enter")
+
+    def code_state_switch():
+        actions.insert("switch () ")
+        actions.key("left")
+        actions.edit.left()
```

Comment:
All of these which put the cursor between the `()` should use `actions.user.insert_between()`.

Reply:
Updated these.